jetty-server/src/main/java/org/eclipse/jetty/server/OptionalSslConnectionFactory.java

```diff
@@ -35,191 +31,76 @@
  * TLS bytes and upgrades to either a TLS connection or to another configurable
  * connection.</p>
  */
-public class OptionalSslConnectionFactory extends AbstractConnectionFactory
+public class OptionalSslConnectionFactory extends DetectorConnectionFactory
```

Comment:
I find it a little surprising that we still need `OptionalSslConnectionFactory` and `ProxyConnectionFactory` as concrete instances of DetectorConnectionFactory.  If they are needed for backwards compatibility then they should be deprecated.

I was expecting to see DetectorConnectionFactory directly used, potentially as part of initialising an AbstractConnector that is given some Detecting factories with no way to select between them.

Is it true that the following are all valid options:
 + SSL or ProxyV1 or ProxyV2
 + (ProxyV1 or ProxyV2) then SSL
 + (ProxyV1 or ProxyV2) then optional SSL
 + SSL then (ProxyV1 or ProxyV2)
 + optional SSL then (ProxyV1 or ProxyV2)

Reply:
The last 2 I would say that don't make much sense, as you want `proxy` to always be the outermost one - although I guess it should be possible to configure `DetectorConnectionFactory` in such way if one needs to.

Reply:
@gregw what you're proposing certainly is possible, it all depends upon the contract we want to have.

Given this comment and the one below, I think the three of us should meet to get that sorted.